java/com/google/flatbuffers/FlexBuffers.java

```diff
@@ -788,7 +788,12 @@ int compareTo(byte[] other) {
                 if (io == other.length) {
```

Comment:
if `other` could have a guaranteed `0` byte in it, this whole `if` could be removed, and above you could just add `|| c2 == '\0'` to get the same effect?

Reply:
but other is an input from user, it has to reallocate the whole length + 1 size and copy. anyways, I'm only updating the condition to avoid any impact.